ImperatorToCK3/Source/Mappers/TagTitleMapper/TagTitleMapper.h

```diff
@@ -0,0 +1,28 @@
+#ifndef TAG_TITLE_MAPPER_H
+#define TAG_TITLE_MAPPER_H
+
+#include "Parser.h"
+#include <set>
+
+namespace mappers
+{
+class TagTitleMapper: commonItems::parser
+{
+  public:
+	void registerTitle(const std::string& impTag, const std::string& ck3Title);
+
+	std::optional<std::string> getTitleForTag(const std::string& impTag);
+
+	[[nodiscard]] const auto& getRegisteredTitleTags() const { return registeredTagTitles; } // used for testing
+
+  private:
+	[[nodiscard]] std::string generateNewTitle(const std::string& impTag) const;
+
+	std::map<std::string, std::string> registeredTagTitles; // We store already mapped countries here.
+	std::set<std::string> usedTitles;
+
+	std::string generatedCK3TitlePrefix = "e_IMPTOCK3_";
```

Comment:
jesus, this crap works? wowza.

Reply:
What do you mean, upper case in titles?

Reply:
no, I meant that x_x_23 and x_mc_48 had some significance - where x_ denoted custom/temporary titles. but you used e_BLAH_XXY and it doesn't mind. Fine by me, just amused.